llvm/lib/Target/AIE/AIE2TargetMachine.cpp

```diff
@@ -64,6 +66,18 @@ static cl::opt<bool> EnablePreMISchedCoalescer(
     "aie-premisched-coalescer", cl::Hidden, cl::init(true),
     cl::desc("Run the coalescer again after the pre-RA scheduler"));
 
+// Option to run internalize pass.
+static cl::opt<bool> InternalizeSymbols(
+    "aie2-internalize-symbols",
+    cl::desc("Enable elimination of non-kernel functions and unused globals"),
+    cl::init(false), cl::Hidden);
+// Option to skip the functions we don't want to internalize.
+static cl::list<std::string>
+    FunctionSkipList("aie2-internalize-skip-functions",
+                     cl::desc("List of function names to skip internalization"),
+                     cl::Hidden, cl::list_init<std::string>({"main"}),
```

Comment:
We sometimes also have a `_main` symbol. I guess we don't have to include it here because `main` is the real entry point?

Reply:
When using the peano flow, the `main` will be the "real" entry point(from today). In our other flow, we have `_main` as the entry point bcoz that's what the linker expects... For internalizing _main, we need to pass it in the command line.